src/pointerwrappers.jl

```diff
@@ -43,8 +43,12 @@ end
 # Non-pointer-type fields get wrapped as a normal PointerWrapper
 PointerWrapper(::Type{T}, pointer) where T = PointerWrapper{T}(pointer)
 
-# Allows to convert a PointerWrapper into type `T`, e.g. convert PointerWrapper{Ptr{Nothing}} in PointerWrapper{Int64}
-PointerWrapper(::Type{T}, pw::PointerWrapper) where T = PointerWrapper{T}(pointer(pw))
+# Allows to convert a PointerWrapper pointing to `Cvoid` into type `T`, e.g. convert PointerWrapper{Ptr{Nothing}} in PointerWrapper{Int64}
+PointerWrapper(::Type{T}, pw::PointerWrapper{Nothing}) where T = PointerWrapper{T}(pointer(pw))
+PointerWrapper(::Type{T}, pw::PointerWrapper{Ptr{Nothing}}) where T = PointerWrapper{T}(pointer(pw))
```

Comment:
does this make sense? Aren't we skipping one level of pointer indirection here? Why isn't it sufficient to just have the first version in the line above?

Same questions apply to the reverse conversion below

Reply:
The first version is not enough, because in Trixi.jl, e.g. [here](https://github.com/trixi-framework/Trixi.jl/pull/1434/files/f92f55c63c765f3c4b7b2776e8c3cbe909c7a11d#diff-51b2f31af1447da121ebcc525279c9c0d65d18d444f84c2a29506ff59e64c673R1610) `info_pw.quad.p.user_data` is of type `PointerWrapper{Ptr{Nothing}}(Ptr{Ptr{Nothing}}`.

Reply:
But shouldn't it be then something like 
```suggestion
PointerWrapper(::Type{T}, pw::PointerWrapper{Ptr{Nothing}}) where T = PointerWrapper{Ptr{T}}(pointer(pw))
```
?

Reply:
Just to make clear: Converting a `PointerWrapper{Nothing}` to `PointerWrapper{Int}` already works by `PointerWrapper(Int, pw_nothing)` if `pw_nothing isa PointerWrapper{Nothing}`. So this PR is obsolete. Sorry for the noise.

Reply:
If you are sure this is not needed anymore, feel free to close the PR.